dotnet/src/webdriver/Firefox/Preferences.cs

```diff
@@ -64,9 +63,31 @@ public Preferences(Dictionary<string, object> defaultImmutablePreferences, Dicti
         /// <param name="value">A <see cref="string"/> value give the preference.</param>
         /// <remarks>If the preference already exists in the currently-set list of preferences,
         /// the value will be updated.</remarks>
+        /// <exception cref="ArgumentNullException">If <paramref name="key"/> or <paramref name="value"/> are <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException">
+        /// <para>If <paramref name="value"/> is wrapped with double-quotes.</para>
+        /// <para>-or-</para>
+        /// <para>If the specified preference is immutable.</para>
+        /// </exception>
         internal void SetPreference(string key, string value)
         {
-            this.SetPreferenceValue(key, value);
+            if (key is null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            if (value is null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            if (IsWrappedAsString(value))
+            {
+                throw new ArgumentException(string.Format(CultureInfo.InvariantCulture, "Preference values must be plain strings: {0}: {1}", key, value));
+            }
+
+            this.ThrowIfPreferenceIsImmutable(key, value);
+            this.preferences[key] = string.Format(CultureInfo.InvariantCulture, "\"{0}\"", value);
```

Comment:
The existing logic wraps strings into json strings, so `abc` becomes `"abc"`. It achieves this by manually adding `"` to the front and end of the string.

This can be achieved simpler by doing `JsonValue.Create(value).ToJsonString()`. Not only will this wrap the string in double-quotes, it will also escape special characters using the format JSON rules. [Example](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8CwAUAAIBMAjEcQAwAEx5AdACoCmAHgowFIDOA9jEYA5fhla8A3ESrkAnAAo+ggGoQANqlaMAwslYQErBQCJAPBuAQfZMBKFv2UwAygmSwA5guvXJQA=)

Should I make this change? This method is publicly reachable by users.

Reply:
Looking through the available types, I found a method which does the JSON string encoding, but does not add the double-quotes. The following would make things type-safe, otherwise users can mess up the JSON.

```suggestion
            this.preferences[key] = string.Format(CultureInfo.InvariantCulture, "\"{0}\"", JsonEncodedText.Encode(value));
```

Reply:
Investigated. We have `webdriver.json` file, bundled as assembly resource. This file contains some default values (key/value pairs, grouped to mutable/frozen). Then at runtime we deserialize it, and provide API to modify key/values in memory. And finally all key/values are going into `*.js` file:
```js
user_pref("key1", value1);
user_pref("key2", value2);
```

where `value` is escaped like `preference.Value.Replace(@"\", @"\\")`.

So, don't apply escaping rules for **json**, I believe it is not aligned with js source code rules. Safer to not touch it.